tedana/viz.py

```diff
@@ -0,0 +1,257 @@
+            ax.axis('off')
+            ax.imshow(np.rot90(ts_B[imgslice * cuts[0], :, :, compnum], k=1),
+                      vmin=imgmin, vmax=imgmax, aspect='equal',
+                      cmap='coolwarm')
+
+            # Second row
+            ax = plt.subplot2grid((5, 6), (2, imgslice - 1), rowspan=1, colspan=1)
+            ax.axis('off')
+            ax.imshow(np.rot90(ts_B[:, imgslice * cuts[1], :, compnum], k=1),
+                      vmin=imgmin, vmax=imgmax, aspect='equal',
+                      cmap='coolwarm')
+
+            # Third Row
+            ax = plt.subplot2grid((5, 6), (3, imgslice - 1), rowspan=1, colspan=1)
+            ax.axis('off')
+            ax_im = ax.imshow(ts_B[:, :, imgslice * cuts[2], compnum],
+                              vmin=imgmin, vmax=imgmax, aspect='equal',
+                              cmap='coolwarm')
+
+        # Add a color bar to the plot.
+        ax_cbar = allplot.add_axes([0.8, 0.3, 0.03, 0.37])
+        cbar = allplot.colorbar(ax_im, ax_cbar)
+        cbar.set_label('Component Beta', rotation=90)
+        cbar.ax.yaxis.set_label_position('left')
+
+        # Get fft and freqs for this subject
+        # adapted from @dangom
+        spectrum, freqs = get_spectrum(mmix[:, compnum], tr)
+
+        # Plot it
+        ax_fft = plt.subplot2grid((5, 6), (4, 0), rowspan=1, colspan=6)
+        ax_fft.plot(freqs, spectrum)
+        ax_fft.set_title('One Sided fft')
+        ax_fft.set_xlabel('Hz')
+        ax_fft.set_xbound(freqs[0], freqs[-1])
+
+        # Fix spacing so TR label does overlap with other plots
+        allplot.subplots_adjust(hspace=0.4)
+        plot_name = 'comp_{}.png'.format(str(compnum).zfill(3))
+        compplot_name = os.path.join('figures', plot_name)
+        plt.savefig(compplot_name)
+        plt.close()
+
+
+def writekappascatter(comptable):
+    """
+    Creates a scatter plot of Kappa vs Rho values. The shape and size of the
+    points is based on classification and variance explained, respectively.
+
+    Parameters
+    ----------
+    comptable : (N x 5) array_like
+        Array with columns denoting (1) index of component, (2) Kappa score of
+        component, (3) Rho score of component, (4) variance explained by
+        component, and (5) normalized variance explained by component
+
+    """
+
+    # Creating Kappa Vs Rho plot
+    ax_scatter = plt.gca()
+
+    # Prebuild legend so that the marker sizes are uniform
+    plt.scatter([], [], s=1, marker='*', c='g', label='accepted', alpha=0.5)
+    plt.scatter([], [], s=1, marker='v', c='r', label='rejected', alpha=0.5)
+    plt.scatter([], [], s=1, marker='d', c='k', label='ignored', alpha=0.5)
+    plt.scatter([], [], s=1, marker='^', c='m', label='midk', alpha=0.5)
+    ax_scatter.legend(markerscale=10)
+
+    # Set up for varying marker shape and color
+    mkr_dict = {'accepted': '*', 'rejected': 'v', 'ignored': 'd', 'midk': '^'}
```

Comment:
We could actually combine these dictionaries, so that each class is associated with a list of `[icon, color]`. Then we could make the `# Prebuild legend section` into a little loop through the dictionary ! Similar to what you have below, but in the same dictionary.

Or, if you think it's clearer as two dictionaries, we could just make the section above (at L175) a loop like the `for kind in mkr_dict` on at L185.

Reply:
Dictionaries combined, loops added